### ISSUE TITLE
False positive for 2018 unused extern crate in global allocator

### ISSUE THREAD

**Issue by Mark-Simulacrum** [rust-lang/rust#52483]
Possibly macro related, not sure.

https://play.rust-lang.org/?gist=7c9846d22f310efa6ab4bec086d61d4e&version=nightly&mode=debug&edition=2018

```rust
#![deny(rust_2018_idioms)]

use alloc_system;

#[global_allocator]
static ALLOC: alloc_system::System = alloc_system::System;

fn main() {}
```

```
error: `extern crate` is not idiomatic in the new edition
 --> src/main.rs:6:1
  |
6 | static ALLOC: alloc_system::System = alloc_system::System;
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: convert it to a `use`
  |
note: lint level defined here
 --> src/main.rs:1:9
  |
1 | #![deny(rust_2018_idioms)]
  |         ^^^^^^^^^^^^^^^^
  = note: #[deny(unused_extern_crates)] implied by #[deny(rust_2018_idioms)]
```

cc @alexcrichton

**Comment by oli-obk** [comment]
Possibly solved by https://github.com/rust-lang/rust/pull/52467 

If not, the function introduced in that PR can be used to solve this

**Comment by alexcrichton** [comment]
I've confirmed this is fixed by #52467
